### PR TITLE
Create taxonomy classes

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
# Description

Create taxonomy classes for RCC Playbook.

Link to sample test page:
Related PRs/issues: #10425 

# Checklist

**Tests**
- [X] Is the code I'm adding covered by tests?

**Changes in Models:**
- [X] Did I update or add new fake data?
- [X] Did I squash my migration?
- [X] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

**Testing**

1. Log into the CMS
2. Go to Snippets
3. Add/remove/edit `RCC Content Types`, `RCC Curricular Areas` and `RCC Topics`
4. Make sure to check ACs in #10425

### CODE REVIEW

File: network-api/networkapi/wagtailpages/factory/libraries/rcc/__init__.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 from networkapi.utility.faker import helpers as faker_helpers
 from networkapi.wagtailpages import models as wagtailpage_models
+
+# from networkapi.wagtailpages.factory.libraries.rcc import relations as relations_factory
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/factory/libraries/rcc/__init__.py:4]: **Change**

Remove commented code

File: network-api/networkapi/wagtailpages/factory/libraries/rcc/taxonomies.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+class RCCTopicFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = wagtailpage_models.RCCTopic
+
+    name = factory.Faker("text", max_nb_chars=25)
+
+    @factory.post_generation
+    def set_slug(obj, created, extracted, **kwargs):
+        obj.slug = text_utils.slugify(obj.name)
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/factory/libraries/rcc/taxonomies.py:37]: **Suggestion**

Could you reduce the code down here by using a BaseFactory? E.G:

```python
class RCCBaseTaxonomyFactory(factory.django.DjangoModelFactory):
    class Meta:
        abstract = True
    
    name = factory.Faker("text", max_nb_chars=25)
    
    @factory.post_generation
    def set_slug(obj, created, extracted, **kwargs):
        obj.slug = text_utils.slugify(obj.name)


class RCCContentTypeFactory(BaseFactory):
    class Meta:
        model = wagtailpage_models.RCCContentType


class RCCCurricularAreaFactory(BaseFactory):
    class Meta:
        model = wagtailpage_models.RCCCurricularArea


class RCCTopicFactory(BaseFactory):
    class Meta:
        model = wagtailpage_models.RCCTopic
```

File: network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/taxonomies.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+@snippet_models.register_snippet
+class RCCCurricularArea(BaseTaxonomy):
+    pass
+
+
+@snippet_models.register_snippet
+class RCCTopic(BaseTaxonomy):
+    pass
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/taxonomies.py:18]: **praise**

This looks neat

File: network-api/networkapi/wagtailpages/tests/libraries/rcc/test_taxonomy.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        # Creating a taxonomy with the same name and slug in a different locale
+        taxonomy_fr = taxonomies_factory.RCCTopicFactory(name="Category", slug="category", locale=self.fr_locale)
+        self.assertIsNone(taxonomy_fr.validate_unique())
+
+        # Creating a taxonomy with the same name and slug in the same locale
+        with self.assertRaises(IntegrityError):
+            taxonomy_duplicate = taxonomies_factory.RCCTopicFactory(
+                name="Category", slug="category", locale=self.default_locale
+            )
+            taxonomy_duplicate.validate_unique()
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/tests/libraries/rcc/test_taxonomy.py:60]: **Notes**

Generally, I think this is ok at this point, testing a single concrete taxonomy that inherits a base is good.
I think if the concrete taxonomies end up diverging slightly then we will need to create similar tests, just something to bare in mind :)

File: network-api/networkapi/wagtailpages/tests/libraries/rcc/test_taxonomy.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        The test creates a RCCTopic instance using the factory with a specific name and slug.
+        It then asserts that the string representation of the instance matches the provided name.
+        """
+        taxonomy = taxonomies_factory.RCCTopicFactory(name="Category", slug="category")
+
+        self.assertEqual(str(taxonomy), "Category")
+
+    def test_validate_unique(self):
+        """
+        Test the uniqueness validation of a RCCTopic instance.
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/tests/libraries/rcc/test_taxonomy.py:38]: **Praise**

...for testing the uniqueness

**Review by kevinhowbrook — APPROVED** [review]
Thanks @jhonatan-lopes - this looks good. Just a few suggestions so I've marked this as approved

File: network-api/networkapi/wagtailpages/factory/libraries/rcc/taxonomies.py (hunk 2 above)
Comment (jhonatan-lopes) [reply network-api/networkapi/wagtailpages/factory/libraries/rcc/taxonomies.py:37]: Good catch!

**Pull request merged**